src/commands/contract/download_contract_abi/mod.rs

```diff
@@ -0,0 +1,126 @@
+
+use crate::common::RpcQueryResponseExt;
+
+#[derive(Debug, Clone, interactive_clap::InteractiveClap)]
+#[interactive_clap(input_context = crate::GlobalContext)]
+#[interactive_clap(output_context = ContractContext)]
+pub struct Contract {
+    #[interactive_clap(skip_default_input_arg)]
+    /// What is the contract account ID?
+    account_id: crate::types::account_id::AccountId,
+    #[interactive_clap(named_arg)]
+    /// Enter the name of the file to save the contract ABI:
+    save_to_file: DownloadContractAbi,
+}
+
+#[derive(Debug, Clone)]
+pub struct ContractContext {
+    global_context: crate::GlobalContext,
+    account_id: near_primitives::types::AccountId,
+}
+
+impl ContractContext {
+    pub fn from_previous_context(
+        previous_context: crate::GlobalContext,
+        scope: &<Contract as interactive_clap::ToInteractiveClapContextScope>::InteractiveClapContextScope,
+    ) -> color_eyre::eyre::Result<Self> {
+        Ok(Self {
+            global_context: previous_context,
+            account_id: scope.account_id.clone().into(),
+        })
+    }
+}
+
+impl Contract {
+    pub fn input_account_id(
+        context: &crate::GlobalContext,
+    ) -> color_eyre::eyre::Result<Option<crate::types::account_id::AccountId>> {
+        crate::common::input_non_signer_account_id_from_used_account_list(
+            &context.config.credentials_home_dir,
+            "What is the contract account ID?",
+        )
+    }
+}
+
+#[derive(Debug, Clone, interactive_clap::InteractiveClap)]
+#[interactive_clap(input_context = ContractContext)]
+#[interactive_clap(output_context = DownloadContractContext)]
+pub struct DownloadContractAbi {
+    #[interactive_clap(skip_default_input_arg)]
+    /// Enter the name of the file to save the contract ABI:
+    file_path: crate::types::path_buf::PathBuf,
+    #[interactive_clap(named_arg)]
+    /// Select network
+    network_config: crate::network_view_at_block::NetworkViewAtBlockArgs,
+}
+
+#[derive(Clone)]
+pub struct DownloadContractContext(crate::network_view_at_block::ArgsForViewContext);
+
+impl DownloadContractContext {
+    pub fn from_previous_context(
+        previous_context: ContractContext,
+        scope: &<DownloadContractAbi as interactive_clap::ToInteractiveClapContextScope>::InteractiveClapContextScope,
+    ) -> color_eyre::eyre::Result<Self> {
+        let on_after_getting_block_reference_callback: crate::network_view_at_block::OnAfterGettingBlockReferenceCallback = std::sync::Arc::new({
+            let account_id = previous_context.account_id.clone();
+            let file_path: std::path::PathBuf = scope.file_path.clone().into();
+
+            move |network_config, block_reference| {
+                if let Ok(contract_abi_response) = tokio::runtime::Runtime::new()
+                    .unwrap()
+                    .block_on(super::inspect_contract::get_contract_abi(&network_config.network_name, &network_config.json_rpc_client(), block_reference, &account_id))
```

Comment:
I would make `get_contract_abi` function return the AbiRoot (for the Ok variant), so you don't need to parse it here.

Reply:
Done